packages/plugin-e2e/src/e2e-selectors/type-testing/types.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * The types in this section are used to contraint the versioned selectors
+ * so they are defined in a consistent way.
+ */
+
+export interface SelectorOptions {}
+
+// We could define this to be something else. Just continuing on the
+// idea I had before. Not important for the overall structure.
+export type SelectorFunction<Type extends SelectorOptions = SelectorOptions> = (options?: Type) => string;
+
+export type VersionedSelector = {
+  [Property: string]: SelectorFunction;
```

Comment:
Should be `[Property: string]: SelectorFunction | string;` so we support string selectors too.